apps/platform/src/components/dashboard/variable/addVariableDialogue/index.tsx

```diff
@@ -129,111 +148,82 @@ export default function AddVariableDialogue() {
-                    {environments.map((env) => (
-                      <SelectItem key={env.id} value={env.slug}>
-                        {env.name}
-                      </SelectItem>
-                    ))}
-                  </SelectContent>
-                </Select>
-              </div>
-
-              <div className="h-[4.5rem] w-[13.375rem] space-y-2">
-                <label
-                  className="h-[1.25rem] w-[9.75rem] text-base font-semibold"
-                  htmlFor="env-value"
-                >
-                  Environment Value
-                </label>
-                <Input
-                  className="h-[2.75rem] w-[13.5rem] border border-white/10 bg-neutral-800 text-gray-300 placeholder:text-gray-500"
-                  id="env-value"
-                  onChange={(e) =>
-                    setNewVariableData({
-                      ...newVariableData,
-                      environmentValue: e.target.value
-                    })
-                  }
-                  placeholder="Environment Value"
-                  value={newVariableData.environmentValue}
-                />
-              </div>
+          {/* Environment Values Section */}
+          <div className="space-y-4">
+            <h3 className="text-base font-semibold">Environment Values</h3>
+            <div className="space-y-4 max-h-[20rem] overflow-y-auto">
+              {environments.map((env) => (
+                <div className="flex items-center gap-4" key={env.slug}>
+                  <label className="sr-only" htmlFor={`env-${env.slug}`}>
+                    {env.name}
+                  </label>
+                  <Input
+                    className="w-[13.5rem] border border-white/10 bg-neutral-800 text-gray-300"
+                    readOnly
+                    value={env.name}
+                  />
+                  <Input
+                    className="w-[13.5rem] border border-white/10 bg-neutral-800 text-gray-300"
+                    id={`env-${env.slug}`}
+                    onChange={(e) => setNewVariableData(prev => ({
+                      ...prev,
+                      environmentEntries: {
+                        ...prev.environmentEntries,
+                        [env.slug]: e.target.value
+                      }
+                    }))}
+                    placeholder={`Value for ${env.name}`}
+                    value={newVariableData.environmentEntries[env.slug] || ''}
+                  />
+                </div>
+              ))}
             </div>
+          </div>
 
-            <div className="flex justify-end pt-4">
-              <Button
-                className="h-[2.625rem] w-[6.25rem] rounded-lg bg-white text-xs font-semibold text-black hover:bg-gray-200"
-                disabled={isLoading}
-                onClick={handleAddVariable}
-              >
-                Add Variable
-              </Button>
-            </div>
+          {/* Submit Button */}
```

Comment:
Instead of this we have a spiral loader. That falls out of the scope of this issue. So you can reset this change